src/model/schema/ThunderstoreSchema.ts

```diff
@@ -0,0 +1,49 @@
+import Ajv from "ajv";
+import addFormats from "ajv-formats";
+
+import { ThunderstoreEcosystem }  from "../../assets/data/ecosystem.d";
+import ecosystem from "../../assets/data/ecosystem.json";
+import jsonSchema from "../../assets/data/ecosystemJsonSchema.json";
+import R2Error from "../errors/R2Error";
+
+
+export class EcosystemSchema {
+    private static _isValidated: boolean = false;
+
+    /**
+     * Get a validated instance of the ecosystem schema.
+     */
+    private static get get(): ThunderstoreEcosystem {
+        if (this._isValidated) {
+            return ecosystem as ThunderstoreEcosystem;
+        }
+
+        // Validate the schema via its schema schema.
+        const ajv = new Ajv();
+        addFormats(ajv);
+
+        const validate = ajv.compile(jsonSchema);
+        const isOk = validate(ecosystem);
+
+        if (!isOk) {
+            throw new R2Error("Schema validation error", ajv.errorsText(validate.errors));
+        }
+
+        this._isValidated = true;
+        return ecosystem as ThunderstoreEcosystem;
+    }
+    /**
+     * Get a list of r2modman entries i.e. games supported by the mod manager.
+     */
+    static get supportedGames() {
+        return Object.values(this.get.games).filter(
+            (x): x is typeof x & {r2modman: NonNullable<typeof x["r2modman"]>} => x.r2modman != null
```

Comment:
Can we make this easier to read?

Reply:
Based on changes in a latter commit, swapped the order of `filter` and `flatMap` and [made the typing IMO easier to read](https://github.com/ebkr/r2modmanPlus/pull/1724/files#diff-42d08a9764b57b0ca2942ed552da05dd1fb87aad9070ed990e7ddf295e550b48R39-R45). I could have imported the R2Modman type from ecosystem type definitions, but I think using `NonNullable` makes the intent clearer, even if it's a bit more verbose.